src/main/scala/viper/gobra/reporting/StatsCollector.scala

```diff
@@ -0,0 +1,376 @@
+      case GobraEntityFailureMessage(taskName, _, e, info, _, time, cached) if !info.node.isInstanceOf[BuiltInMember] =>
+        handleEntityMessage(taskName, e, info, time, cached, success = false)
+      case _ =>
+    }
+    // Pass message to next reporter
+    reporter.report(msg)
+  }
+
+  def getNumberOfCachedViperMembers: Int = memberMap.values.flatMap(_.viperMembers.values).count(_.cached)
+
+  def getNumberOfCacheableViperMembers: Int = memberMap.values.flatMap(_.viperMembers.values).count(member => member.hasBody && member.nodeType == ViperNodeType.Method)
+  def writeJsonReportToFile(file: File): Unit = {
+    if((file.exists() && file.canWrite) || file.getParentFile.canWrite) {
+      FileUtils.writeStringToFile(file, getJsonReport, UTF_8)
+    }
+  }
+
+  def getJsonReport: String = {
+    val memberJson = memberMap.values.map(_.asJson(i)).mkString(", \n")
+    val json = s"""[
+      |$memberJson
+      |]""".stripMargin
+    json
+  }
+
+  def findGobraMemberByViperMemberName(taskName: String, viperMemberName: String, pkgDir: String): Option[GobraMemberEntry] =
+    viperMemberNameGobraMemberMap.get(viperMemberKey(taskName, viperMemberName, pkgDir))
+      .orElse(viperMemberNameGobraMemberMap.get(viperMemberKey(taskName, viperMemberName)))
+
+  def getWarnings(pkgDir: String, pkg: String, config: Config): Set[String] =
+    memberMap.values
+      .filter(gobraMember => gobraMember.pkgDir == pkgDir && gobraMember.pkg == pkg)
+      .flatMap(gobraMember => {
+        val name = gobraMember.pkg + "." + gobraMember.memberName + gobraMember.args
+        gobraMember.dependencies().flatMap({
+          // Trusted implies abstracted, so we match trusted first
+          case GobraMemberEntry(_, pkg, memberName, _, args, _, true, _) =>
+            Some("Warning: Member " + name + " depends on trusted member " + pkg + "." + memberName + args + "\n")
+          case GobraMemberEntry(_, pkg, memberName, _, args, _, _, true) =>
+            Some("Warning: Member " + name + " depends on abstract member " + pkg + "." + memberName + args + "\n")
+          case GobraMemberEntry(pkgDir, pkg, _, _, _, _, _, _) if !config.inputPackageMap.contains(PackageEntry(pkgDir, pkg)) =>
+            Some("Warning: Depending on imported package that is not verified: " + pkgDir + " - " + pkg)
+          case _ => None
+        })
+      }).toSet
+
+  def addResult(pkgDir: String,
+                pkg: String,
+                memberName: String,
+                args: String,
+                nodeType: GobraNodeType,
+                viperMember: ViperMemberEntry,
+                isTrusted: Boolean,
+                isAbstract: Boolean): Unit = {
+    val key = gobraMemberKey(pkgDir, pkg, memberName, args)
+
+    val viperKey = viperMemberKey(viperMember.taskName, viperMember.memberName, pkgDir)
+    val shortViperKey = viperMemberKey(viperMember.taskName, viperMember.memberName)
```

Comment:
why is it necessary to always perform two lookups, once with the long and once with the short key? Are there Viper members for which wo do not know to which package they belong?

Reply:
This is because of the ambiguous naming described in https://github.com/viperproject/gobra/issues/417. because of it, the assumption that a viper member can only belong to a single GobraMember is violated, which is why I resorted to this hack. There also is a comment about this in  `addResult`

Reply:
I think that would simplify the code and I'm not so keen on having additional code just as an ugly workaround for a bug that should be quite easy to fix

Reply:
There actually is such a case in VerifiedSCION.That is why I became aware of this issue.
But I think if it's only for the purpose of evaluation, I could as well just rename the package in VerifiedSCION, then we don't have to do this

Reply:
Couldn't we instead already fix issue 417 by using the package path instead of the package name?

Reply:
Good point, maybe after this PR I could prepare a PR that removes this hack and add a note that it is to be merged after the issue is resolved?

Reply:
> Couldn't we instead already fix issue 417 by using the package path instead of the package name?

Would also be an alternative

Reply:
> > Couldn't we instead already fix issue 417 by using the package path instead of the package name?
> 
> Would also be an alternative

That should be the easiest solution. It's only about adapting the `name` and `nameWithoutScope` functions in the NameManager class (in Desugarer.scala). Replace the package name by a unique package identifier